src/Compilers/CSharp/Portable/Syntax/SyntaxExtensions.cs

```diff
@@ -226,6 +206,25 @@ internal static SyntaxNode SkipParens(this SyntaxNode expression)
             return expression;
         }
 
+        private static bool IsDeconstructionDeclaration(this ExpressionSyntax self)
```

Comment:
>IsDeconstructionDeclaration [](start = 28, length = 27)

Consider using a different name because this methods overload each other inside this class, but they will do different things and produce different results if both are called with a AssignmentExpressionSyntax node.